[rstmgr] Update reset info to include software reset (#10428)
lowRISC/opentitan: hw/ip/rstmgr/doc/_index.md

`POR`                   | Cold boot, the system was reset through POR circuitry.
`LOW_POWER_EXIT`        | Warm boot, the system was reset through low power exit.
`NDM RESET`             | Warm boot, the system was reset through `rv_dm` non-debug-module request.
`SW_REQ`                | Warm boot, the system was reset through {{< regref "RST_REQ" >}}.
`HW_REQ`                | Warm boot, the system was reset through peripheral requests.  There may be multiple such requests.

The reset info register is write 1 clear.
It is software responsibility to clear old reset reasons; the reset manager simply records based on the rules below.

Excluding power on reset, which is always recorded when the device POR circuitry is triggered, the other resets are recorded when authorized by the reset manager.
Reset manager authorization is based on reset categories as indicated by the power manager.
The power manager observes three categories of reset states that are mutually exclusive.

> *   No reset triggered through the power manager.

Reviewer: This is rather awkward: you are saying that power manager observes these, but one of them is not even seen by pwrmgr. Maybe you mean "The reset manager observes...".

Also, the first bullet item can be "Reset not triggered through the power manager."

Author: sure...

*   Reset triggered by low power entry.
*   Reset triggered by software or a peripheral request.

Whenever the power manager begins one of the latter two sequences, it sends a hint to the reset manager so that the reset manager can decide which reason to record when the processor reset is observed.

> Whenever the power manager is NOT in one of the two latter states, non-debug-module resets are allowed and directly recorded by the reset manager.

Reviewer: Here you are explicitly saying the resets not triggered through power manager are non-debug module resets. Perhaps you can make that explicit in that bullet item instead.

Author: sure...

Author: no that is not what i am saying.

`POR`                   | Cold boot, the system was reset through POR circuitry.
`LOW_POWER_EXIT`        | Warm boot, the system was reset through low power exit.
`NDM RESET`             | Warm boot, the system was reset through `rv_dm` non-debug-module request.
`SW_REQ`                | Warm boot, the system was reset through {{< regref "RST_REQ" >}}.
`HW_REQ`                | Warm boot, the system was reset through peripheral requests.  There may be multiple such requests.

The reset info register is write 1 clear.
It is software responsibility to clear old reset reasons; the reset manager simply records based on the rules below.

Excluding power on reset, which is always recorded when the device POR circuitry is triggered, the other resets are recorded when authorized by the reset manager.
Reset manager authorization in turn is based on reset categories as indicated by the power manager.

> The power manager observes 4 categories of reset states that are mutually exclusive.

Reviewer: ```suggestion
The power manager observes four categories of reset states that are mutually exclusive.
```
To be consistent with the text following.

> *   No resets are triggered through the power manager.

Reviewer: I don't see why "No " here. I assume this refers to resets that are triggered by pwrmgr, like power glitch or escalation.

Author: i hvae no idea what i was trying to say there. Let me reword.

Author: o i remember now, i was trying to say one of the cases is that no resets are triggered through the pwrmgr (ndm_rst)

*   Resets triggered by low power entry.
*   Resets triggered by software.
*   Resets triggered by a peripheral request.

Whenever the power manager begins one of the latter three sequences, it sends a hint to the reset manager so that the reset manager can decide which reason to record when the processor reset is observed.

> Whenever the power manager is NOT in one of the two latter states, non-debug-module resets are allowed and directly handled and recorded by the reset manager.

Reviewer: ```suggestion
Whenever the power manager is NOT in one of the three latter states, non-debug-module resets are allowed and directly handled and recorded by the reset manager.
```

Since a reset could be motivated by multiple reasons (a security escalation during low power transition for example), the reset information registers constantly record all reset causes in which it is allowed.
The only case where this is not done is `POR`, where active recording is silenced until the first processor reset release.

> Despite 4 reset causes all labeled as warm boot, their effects on the system are not identical.

Reviewer: ```suggestion
Despite four reset causes are labeled as warm boot, their effects on the system are not identical.
```
